### PR TITLE
Add partners resource

### PR DESCRIPTION
**Pull request by 0urobor0s**
Created the partner entity and all of it's attributes

### CODE REVIEW

**Comment by 0urobor0s** [comment]
I thought that it was more simple, to only add the model
Will look on the updated issue

File: app/controllers/partners_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  def edit
+  end
+
+  def create
+    @partner = Partner.new(partner_params)
+
+    if @partner.save
+      #redirect_to @partner
```
Comment (marcelo140) [inline app/controllers/partners_controller.rb]: Delete this line.

File: app/views/partners/_form.slim [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    class: "partner-form-label"
+  br
+  = form.text_area :link,
+    class: "partner-form-field"
+  br
+  = form.label :logo,
+    class: "partner-form-label"
+  br
+  = form.file_field :logo,
+    class: "partner-form-field"
```
Comment (marcelo140) [inline app/views/partners/_form.slim]: Just asking to be sure. Do the field's names appear capitalized?

File: app/views/partners/index.slim [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+.divide40
+.container
+
+  - if can_create?
+    = link_to 'Criar Parceria', new_partner_path,
+      class: "btn btn-lg border-theme"
+  hr
+
+  - if @partners.empty?
+    h4 Parcerias não existentes
```
Comment (marcelo140) [inline app/views/partners/index.slim]: Can we change this to a friendlier message? Something like "De momento ainda não temos parcerias, volta aqui mais tarde"

File: app/controllers/partners_controller.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
+class PartnersController < ApplicationController
+  before_action :set_partner, only: [:show, :edit, :update, :destroy]
```
Comment (marcelo140) [inline app/controllers/partners_controller.rb]: You don't have a `show` method.

File: app/views/partners/show.slim [hunk 5]
```diff
@@ -0,0 +1,13 @@
+.container
```
Comment (marcelo140) [inline app/views/partners/show.slim]: You don't have a `show` method.

File: config/routes.rb [hunk 6]
```diff
@@ -35,11 +35,12 @@
     resources :terms,       only: [:new, :create, :destroy]
   end
 
+  resources :partners,      only: [:index, :new, :edit, :create, :update, :destroy]
```
Comment (marcelo140) [inline config/routes.rb]: You could just write `resources :partners, except: :show`

File: db/migrate/20171031225933_create_partners.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class CreatePartners < ActiveRecord::Migration
+  def change
+    create_table :partners do |t|
+      t.string :name, limit: 75
+      t.string :benefits
```
Comment (marcelo140) [inline db/migrate/20171031225933_create_partners.rb]: Since this attribute can hold more details than simple benefits, can we change this to description?

File: app/helpers/partners_helper.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+module PartnersHelper
+  def can_create?
+    can? :create, Partner
+  end
+
+  def error_messages(partner)
+    partner.errors.full_messages
+  end
+end
```
Comment (marcelo140) [inline app/helpers/partners_helper.rb]: Both these methods are quite simple, I don't really think the helpers are needed.

File: app/controllers/partners_controller.rb [hunk 9]
```diff
@@ -0,0 +1,46 @@
+class PartnersController < ApplicationController
+  before_action :set_partner, only: :edit
```
Comment (marcelo140) [inline app/controllers/partners_controller.rb]: I don't understand why did you add this again. Specially for a method that you accidently deleted x)

File: app/views/partners/_form.slim (hunk 2 above)
Comment (marcelo140) [reply app/views/partners/_form.slim]: bump on this

File: app/controllers/partners_controller.rb [hunk 10]
```diff
@@ -0,0 +1,46 @@
+class PartnersController < ApplicationController
```
Comment (marcelo140) [inline app/controllers/partners_controller.rb]: You need to setup the cancancan as described in their [docs](https://github.com/CanCanCommunity/cancancan/wiki/Authorizing-controller-actions) so that the actions are correctly restricted.

File: app/models/ability.rb [hunk 11]
```diff
@@ -30,5 +30,6 @@ def activity_admin_permissions(user)
     user_permissions(user)
     can :manage, Activity
     can [:index, :update], Registration
+    can :manage, Partner
```
Comment (marcelo140) [inline app/models/ability.rb]: Guests should be able to use the `index` action.

**Comment by marcelo140** [comment]
ping @0urobor0s @ZePedroResende

File: app/views/partners/index.slim [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  hr
+
+  - if @partners.empty?
+    h4 De momento ainda não temos parcerias, volta aqui mais tarde
+
+  existentes
+  - @partners.each do |partner|
+    .row
+      - if can_create?
+        = link_to 'Editar', edit_partner_path(partner),
```
Comment (marcelo140) [inline app/views/partners/index.slim]: I don't like the edit button positioning. It moves too much content around in a way that an admin can't rely on this page to see how users see the it.

File: app/views/partners/index.slim [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+          class: "btn btn-lg border-theme"
+      .col-md-2.col-sm-12
+        = link_to image_tag(partner.logo.url, class: "img-responsive activity-poster"), partner.link, target: "_blank"
```
Comment (marcelo140) [inline app/views/partners/index.slim]: When a partner has no image the layout deteriorates and instead appears an useless `missing` link. We need a different solution for this.

File: app/views/partners/index.slim [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+      .col-md-10.col-sm-12
+        h4.partner-name
```
Comment (marcelo140) [inline app/views/partners/index.slim]: I have a feeling the logo and the text aren't correctly aligned.

File: app/views/partners/index.slim [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+          = link_to partner.name, partner.link, target: "_blank"
+        ul.partner-advantages
+          .partners-list-item
+            = to_markdown(partner.description)
```
Comment (marcelo140) [inline app/views/partners/index.slim]: Both the white circle and the margin before the circles are super ugly :(

File: app/models/ability.rb [hunk 16]
```diff
@@ -29,5 +30,6 @@ def activity_admin_permissions(user)
     user_permissions(user)
     can :manage, Activity
     can %i[index update], Registration
+    can :manage, Partner
```
Comment (marcelo140) [inline app/models/ability.rb]: We better plan how will we handle permissions.
